src/ray/common/buffer.h

```diff
@@ -0,0 +1,34 @@
+#ifndef RAY_COMMON_IO_H
+#define RAY_COMMON_IO_H
+
+#include <cstdint>
+#include <cstdio>
+
+namespace ray {
+
+/// The interface that represents a buffer of bytes.
+class Buffer {
+ public:
+  /// Pointer to the data.
+  virtual uint8_t *Data() = 0;
+
+  /// Length of this buffer.
+  virtual size_t Length() = 0;
```

Comment:
It's better to name this `Size()` to follow STL style?

Reply:
I feel both are fine. I can change it.